### PR TITLE
add zotino example code

### PR DESCRIPTION
**Pull request by mntng**

### CODE REVIEW

File: artiq/gateware/spi.py [hunk 1]
```diff
@@ -70,9 +68,7 @@ class SPIMaster(Module):
           completed transfer.
         * If desired, change xfer register for the next transfer.
         * If desired, write data queuing the next (possibly chained) transfer.
-
```
Comment (sbourdeauducq) [inline artiq/gateware/spi.py]: Why remove those lines?

File: artiq/gateware/spi.py [hunk 2]
```diff
@@ -368,4 +362,4 @@ def __init__(self, t):
     # print(convert(dut))
 
     Tristate.lower = _TestTristate
-    run_simulation(dut, _test_gen(dut.bus), vcd_name="spi_master.vcd")
+    run_simulation(dut, _test_gen(dut.bus), vcd_name="spi_master.vcd")
```
Comment (sbourdeauducq) [inline artiq/gateware/spi.py]: Please look at the patch and remove things like this.

File: artiq/examples/master/zotino.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+class ZotinoTestDAC(EnvExperiment):
+    def build(self):
+        self.setattr_device("core")
+        self.setattr_device("dac_zotino")
+        self.setattr_device("latch_config")
+        self.setattr_device("clk_config")
+        self.setattr_device("ser_config")
+		
+    @kernel
+    def dir_config(self, rclk, srclk, ser, data):
```
Comment (sbourdeauducq) [inline artiq/examples/master/zotino.py]: I would call those consistently with ``io_config`` and with better names, e.g. ``clk``, ``ser_data``, ``latch``.

File: artiq/examples/master/zotino.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        self.setattr_device("ser_config")
+		
+    @kernel
+    def dir_config(self, rclk, srclk, ser, data): 
+        dt = 10*us
+        srclk.off()
+        rclk.off()
+        delay(dt)
+        x = 0x80000000
+        for i in range(32):
```
Comment (sbourdeauducq) [inline artiq/examples/master/zotino.py]: There are other shift registers in the system, e.g. for the Novogorny PGIA, with other lengths. It makes sense to make this function generic to transfer any number of bits (not just 32), document it, and move it into some library (e.g. ``artiq.coredevice.shiftreg``).

Comment (sbourdeauducq) [reply artiq/examples/master/zotino.py]: And this function could be turned into a device that uses and depends on the TTL devices, just like the DAC driver that depends on a SPI device for the data and a TTL device for LDAC.

File: artiq/examples/master/device_db.py [hunk 5]
```diff
@@ -144,44 +144,71 @@
...
+        "class": "TTLOut",
+        "arguments": {"channel": 28}
+    },
+    "io_config_ser": {
+        "type": "local",
+        "module": "artiq.coredevice.ttl",
+        "class": "TTLOut",
+        "arguments": {"channel": 29}
+    },
+    "shift_reg": {
```
Comment (sbourdeauducq) [inline artiq/examples/master/device_db.py]: those device names could be improved.

File: artiq/examples/master/zotino.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+class ZotinoTestDAC(EnvExperiment):
+    def build(self):
+        self.setattr_device("core")
+        self.setattr_device("shift_reg")
+        self.setattr_device("dac_zotino")
+
+    @kernel
+    def run(self):
+        self.core.reset()
+        self.fmc_io_config(self, 0x00008800) ## set lvds direction on fmc
```
Comment (sbourdeauducq) [inline artiq/examples/master/zotino.py]: That's not going to work.

File: artiq/coredevice/shiftreg.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    def __init__(self, dmgr, clk, ser, latch, n=32, dt=10*us):
+        self.core = dmgr.get("core")
+        self.bus = dmgr.get(latch)
+        self.srclk = dmgr.get(clk)
+        self.ser = dmgr.get(ser)
+        self.n = n
+        self.dt = dt
+
+    @kernel
+    def fmc_io_config(self, data):
```
Comment (sbourdeauducq) [inline artiq/coredevice/shiftreg.py]: Shift register + latches are not specific to FMC nor configuring I/O.

File: artiq/examples/master/zotino.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+class ZotinoTestDAC(EnvExperiment):
+    def build(self):
+        self.setattr_device("core")
+        self.setattr_device("shift_reg")
+        self.setattr_device("dac_zotino")
+
+    @kernel
+    def run(self):
+        self.core.reset()
+        self.shift_reg.shiftreg_config(self, 0x00008800) ## set lvds direction on fmc
```
Comment (sbourdeauducq) [inline artiq/examples/master/zotino.py]: Why have ``self`` as parameter?

File: doc/manual/core_device.rst [hunk 9]
```diff
@@ -86,10 +92,16 @@ The board has RTIO SPI buses mapped as follows:
...
 +--------------+--------------+--------------+--------------+------------+
-| 27           | ZOTINO_CS_N  | ZOTINO_MOSI  | ZOTINO_MISO  | ZOTINO_CLK |
+| 30           | ZOTINO_CS_N  | ZOTINO_MOSI  | ZOTINO_MISO  | ZOTINO_CLK |
 +--------------+--------------+--------------+--------------+------------+
 
-The DDS bus is on channel 29.
+The DDS bus is on channel 32.
+
+The Zotino(3U DAC rev1.0) is connected to kc705 FMC HPC ports using VHDCI board(VHDCI breakout rev1.0) to FMC card(FMC DIO ch32 lvds a v1.2).
+On the VHDCI board, J44A is connected to J41 since LVDS2 is now chosen to be connected to J37 on the zotino.
```
Comment (sbourdeauducq) [inline doc/manual/core_device.rst]: This is not clear. It sounds like both J44A and J41 are on the same board. And which "VHDCI board", the FMC/VHDCI or the VHDCI breakout?

Comment (jordens) [reply doc/manual/core_device.rst]: "Name ambiguity due to encoding metadata in the name" strikes again. Nice indication that opaque code names are the way to go and that any approach to make names "intuitive" leads to actual issues with inconsistency, misunderstanding that are way more tangible and hard to relevant than merely complaints *à la* "can't remember the name".

File: doc/manual/core_device.rst [hunk 10]
```diff
@@ -97,7 +97,9 @@ The board has RTIO SPI buses mapped as follows:
 
 The DDS bus is on channel 32.
 
+
 This configuration supports a Zotino connected to the KC705 FMC HPC through a FMC DIO 32ch LVDS v1.2 and a VHDCI breakout board rev 1.0.
+The KC705 FMC HPC should be connected to J44A and the Zotino should be connected to J41 of the VHDCI breadout board.
```
Comment (sbourdeauducq) [inline doc/manual/core_device.rst]: breakout

File: artiq/coredevice/shiftreg.py [hunk 11]
```diff
@@ -1,3 +1,16 @@
+<<<<<<< HEAD
```
Comment (sbourdeauducq) [inline artiq/coredevice/shiftreg.py]: Please look at the patches you are proposing.

File: artiq/coredevice/shiftreg.py [hunk 12]
```diff
@@ -11,10 +24,30 @@ def __init__(self, dmgr, clk, ser, latch, n=32, dt=10*us):
         self.clk = dmgr.get(clk)
         self.ser = dmgr.get(ser)
         self.latch = dmgr.get(latch)
+>>>>>>> 8407b2c400f4c4565f2504d7bc2b6a8ee70e6918
```
Comment (sbourdeauducq) [inline artiq/coredevice/shiftreg.py]: ditto

File: doc/manual/core_device.rst [hunk 13]
```diff
@@ -97,7 +97,9 @@ The board has RTIO SPI buses mapped as follows:
 
 The DDS bus is on channel 32.
 
+
 This configuration supports a Zotino connected to the KC705 FMC HPC through a FMC DIO 32ch LVDS v1.2 and a VHDCI breakout board rev 1.0.
+The KC705 FMC HPC should be connected to J44A and the Zotino should be connected to J41 of the VHDCI breakout board.
```
Comment (sbourdeauducq) [inline doc/manual/core_device.rst]: The FMC breakout board does not have a "J44A" marking on its silkscreen and even less on its future front panel, so referencing it does not make the reader's life easier. You should write this documentation for someone who is setting up their boards the first time, and you should have said the *bottom* connector.

**Pull request closed without merging**